src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -145,6 +151,8 @@ export class DatasetService {
       parent: dataStructure.parent?.id || '',
       children: dataStructure.children?.map((child) => child.id) || [],
       hasNext: dataStructure.hasNext,
+      nextToken: dataStructure.nextToken,
```

Comment:
`hasNext` and `nextToken` are a little confusing. can we callit something that more generic that can be used for other pagination api's too like `paginationToken` or `cursor` like graphql?

Reply:
+1